src/cart/crt_init.c

```diff
@@ -471,9 +471,9 @@ prov_settings_apply(bool primary, crt_provider_t prov, crt_init_options_t *opt)
 	int	rc = 0;
 
 	/* rxm and verbs providers only works with regular EP */
-	if ((prov == CRT_PROV_OFI_VERBS_RXM ||
-	     prov == CRT_PROV_OFI_TCP_RXM) &&
-	    crt_provider_is_sep(prov, primary)) {
+	if (prov != CRT_PROV_OFI_PSM2 &&
+	    prov != CRT_PROV_OFI_SOCKETS &&
+	    crt_provider_is_sep(primary, prov)) {
```

Comment:
wow, wrong order of arguments and it just compiles, gotta love C sometimes

Reply:
Yes, the call was wrong prior to this patch. It was helpful that a failure on this patch led to Alex identifying the mistake.

Reply:
@frostedcmos reminds me of the infamous rep vs req case.